arch/risc-v/src/esp32c3/esp32c3_rtc_lowerhalf.c

```diff
@@ -0,0 +1,575 @@
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+#include <nuttx/spinlock.h>
+
+#include <sys/types.h>
+#include <stdbool.h>
+#include <string.h>
+#include <errno.h>
+
+#include <nuttx/arch.h>
+#include <nuttx/timers/rtc.h>
+
+#include "esp32c3_rtc.h"
+#include "hardware/esp32c3_tim.h"
+
+#ifdef CONFIG_RTC_DRIVER
+
+/****************************************************************************
+ * Private Types
+ ****************************************************************************/
+
+#ifdef CONFIG_RTC_ALARM
+struct esp32c3_cbinfo_s
+{
+  volatile rtc_alarm_callback_t cb;  /* Callback when the alarm expires */
+  volatile FAR void *priv;           /* Private argurment to accompany callback */
+};
+#endif
+
+/* This is the private type for the RTC state.  It must be cast compatible
+ * with struct rtc_lowerhalf_s.
+ */
+
+struct esp32c3_lowerhalf_s
+{
+  /* This is the contained reference to the read-only, lower-half
+   * operations vtable (which may lie in FLASH or ROM)
+   */
+
+  FAR const struct rtc_ops_s *ops;
+#ifdef CONFIG_RTC_ALARM
+  /* Alarm callback information */
+
+  struct esp32c3_cbinfo_s cbinfo[RTC_ALARM_LAST];
+#endif
+};
+
+/****************************************************************************
+ * Private Function Prototypes
+ ****************************************************************************/
+
+/* Prototypes for static methods in struct rtc_ops_s */
+
+static int rtc_rdtime(FAR struct rtc_lowerhalf_s *lower,
+                      FAR struct rtc_time *rtctime);
+static int rtc_settime(FAR struct rtc_lowerhalf_s *lower,
+                       FAR const struct rtc_time *rtctime);
+static bool rtc_havesettime(FAR struct rtc_lowerhalf_s *lower);
+
+#ifdef CONFIG_RTC_ALARM
+static void rtc_alarm_callback(FAR void *arg, unsigned int alarmid);
+static int rtc_setalarm(FAR struct rtc_lowerhalf_s *lower,
+                        FAR const struct lower_setalarm_s *alarminfo);
+static int rtc_setrelative(FAR struct rtc_lowerhalf_s *lower,
+                           FAR const struct lower_setrelative_s *alarminfo);
+static int rtc_cancelalarm(FAR struct rtc_lowerhalf_s *lower,
+                           int alarmid);
+static int rtc_rdalarm(FAR struct rtc_lowerhalf_s *lower,
+                       FAR struct lower_rdalarm_s *alarminfo);
```

Comment:
Could you please rename these operations to use `esp32c3_rtc_lh_` as the prefix, instead of `rtc_`?
Reason:
NuttX coding style determines the functions' name should be prefixed with the module it is.
PTAL: https://nuttx.apache.org/docs/latest/contributing/coding_style.html#function-naming

Reply:
Regarding the issue of prefixes, I have discussed with @gustavonihei , adding `esp32c3_ ` in to a function name is redundant, and using the prefix of `rtc` to describe the function module more accurately, maybe we can discuss it again.

Reply:
According to the discussion, I have changed the `rtc_` prefix to `rtc_lh_`